### ISSUE TITLE
Guardians - Denjin Makai II - Boss face glitch

### ISSUE THREAD

**Issue by BilonFullHDemon** [finalburnneo/FBNeo#1853]
When you grab 1st Boss with Zeldia his face is glitched
![DenjinMakaiII](https://github.com/user-attachments/assets/b6b066ad-150a-44e7-aee5-ae40e44a5699)

**Comment by dinkc64** [comment]
I can't seem to influence this glitch at all :/

**Comment by dinkc64** [comment]
Can you find footage where this glitch doesn't happen?

**Comment by dinkc64** [comment]
I've dug into this quite a bit:
I'm pretty sure its caused by a bug in the game, and not the emulation

If you can find pcb footage where this doesn't happen, I'll certainly throw some more time at it!

best regards,
- dink

**Comment by BilonFullHDemon** [comment]
Hi dinkc64, it seams you are right, it is a bug in the game. is there any chance to provide me with some info so I could fix it? Thank you!

**Comment by dinkc64** [comment]
All I can remember is that it's a: // "normal" sprite
(look in d_seta2.cpp for that line)

best regards,
- dink

**Comment by kokerkov** [comment]
maybe it's normal. I got same result from MAME.

**Comment by leoxxx** [comment]
![Image](https://github.com/user-attachments/assets/674c3f25-d25f-4de8-a319-4a7a63900457)
This bug appears when 1st boss is catched. All heros, not only Zeldia.
@BilonFullHDemon
Did you fix it?

**Comment by dinkc64** [comment]
leoxxx, it's a bug in the game itself

**Comment by leoxxx** [comment]
Yes, it is as you said. [BilonFullHDemon](https://github.com/BilonFullHDemon) said he want to fix it by himself, so I want to know if it is fixed. I think IPS maybe fix it.

**Comment by BilonFullHDemon** [comment]
[BOSS FACE FIX.zip](https://github.com/user-attachments/files/30261033/BOSS.FACE.FIX.zip)

@leoxxx The first boss’s face fix has been implemented in the ROM file.

Unfortunately, I lost the relevant information, including the offset and description. It can be recovered by comparing the files in a hex editor.

As far as I remember, the bug was caused by incorrect texture coordinates.

I also recall several other graphical bugs involving certain weapons during some character animations.

Overall, while investigating this bug, I came to the conclusion that this title was not completely finished. There are placeholders for additional characters, as well as character voice sound effects.

Thank you!

**Comment by leoxxx** [comment]
> [BOSS FACE FIX.zip](https://github.com/user-attachments/files/30261033/BOSS.FACE.FIX.zip)
> 
> [@leoxxx](https://github.com/leoxxx) The first boss’s face fix has been implemented in the ROM file.
> 
> Unfortunately, I lost the relevant information, including the offset and description. It can be recovered by comparing the files in a hex editor.
> 
> As far as I remember, the bug was caused by incorrect texture coordinates.
> 
> I also recall several other graphical bugs involving certain weapons during some character animations.
> 
> Overall, while investigating this bug, I came to the conclusion that this title was not completely finished. There are placeholders for additional characters, as well as character voice sound effects.
> 
> Thank you!


That's great! THX. 
I will try to creat an IPS out of your file.

**Comment by BilonFullHDemon** [comment]
Boss Face fix info:
u5.bin
OFFSET 0x47212 - 0B E5 -> 04 EC

OFFSET 0x472AE - 0B E5 -> 04 EC

BONUS:
JINREI's move: Down, Up + Attack Button, has a mechanic that randomly adds special visual effect to it.
Patch turns randomizer OFF. Special effect is applied every time move is done.
u2.bin
OFFSET 0x535A2 -> FF

Your Welcome! :)

**Comment by dinkc64** [comment]
@BilonFullHDemon that's so elite!
are you going to release a hack, so we can have a clone of the rom with your special version? :)

**Comment by leoxxx** [comment]
JINREI, ZELDIA, Girulian and Kurokishi have the bug. Others haven't the bug.

**Comment by leoxxx** [comment]
> Boss Face fix info: u5.bin OFFSET 0x47212 - 0B E5 -> 04 EC
> 
> OFFSET 0x472AE - 0B E5 -> 04 EC
> 
> BONUS: ROU's move: Down, Up + Attack Button, has a mechanic that randomly adds special visual effect to it. Patch turns randomizer OFF. Special effect is applied every time move is done. u2.bin OFFSET 0x535A2 -> FF
> 
> Your Welcome! :)

It's JINREI's move, not ROU's

**Comment by BilonFullHDemon** [comment]
> > Boss Face fix info: u5.bin OFFSET 0x47212 - 0B E5 -> 04 EC
> > OFFSET 0x472AE - 0B E5 -> 04 EC
> > BONUS: ROU's move: Down, Up + Attack Button, has a mechanic that randomly adds special visual effect to it. Patch turns randomizer OFF. Special effect is applied every time move is done. u2.bin OFFSET 0x535A2 -> FF
> > Your Welcome! :)
> 
> It's JINREI's move, not ROU's

Yeah I just realised that and came here to fix this but I see you figured it out :)

Sorry!

**Comment by BilonFullHDemon** [comment]
> JINREI, ZELDIA, Girulian and Kurokishi have the bug. Others haven't the bug.

1st Boss Face Glitch?

**Comment by BilonFullHDemon** [comment]
[Guardians - DENJIN MAKAI II Eng. Files.zip](https://github.com/user-attachments/files/30623918/Guardians.-.DENJIN.MAKAI.II.Eng.Files.zip)



> [@BilonFullHDemon](https://github.com/BilonFullHDemon) that's so elite! are you going to release a hack, so we can have a clone of the rom with your special version? :)

I was about to, but I only completed the English version and then stopped working on it.

Here is the English hack I made in 2024, I think—I don’t remember exactly.

It contains some other improvements, such as shadows and corrected test-screen text (“SALF” → “SELF”).

Thank you!

**Comment by leoxxx** [comment]
> > JINREI, ZELDIA, Girulian and Kurokishi have the bug. Others haven't the bug.
> 
> 1st Boss Face Glitch?

Yes. The others' grabbing(holding?) moves are a bit different.

**Comment by BilonFullHDemon** [comment]
> > > JINREI, ZELDIA, Girulian and Kurokishi have the bug. Others haven't the bug.
> > 
> > 
> > 1st Boss Face Glitch?
> 
> Yes. The others' grabbing(holding?) moves are a bit different.

Its not a glitch.
Different characters use different texture of this enemy when he is in  garb state.
My fix fixed the texture itself.
I attach image how it was made by the developers.

<img width="1216" height="928" alt="Image" src="https://github.com/user-attachments/assets/66cbd83e-05f6-4a85-b283-77691e0eebe0" />

**Comment by leoxxx** [comment]
grab state?
I can't understand why the object is like this.

**Comment by BilonFullHDemon** [comment]
> grab state? I can't understand why the object is like this.

This is an oversight made by the developers.

Texture coordinates of part that is off where wrong and left for the release build of the game.

Grab state is when your character grab an enemy, then the enemy is in grab state, which triggers an coded to this state texture of the enemy.
